index.js

```diff
@@ -17,6 +38,11 @@ function plugin(settings) {
     function transformer(tree) {
         visit(tree, 'text', function(node) {
             node.value = node.value.replace(RE_EMOJI, getEmoji);
+            
+            if (!RE_EMOJI.test(node.value) && emoticonEnable) {
```

Comment:
```suggestion
            before = node.value
            node.value = before.replace(RE_EMOJI, getEmoji);
            
            if (emoticonEnable && node.value === before) {
```

This request includes below two points:

1. `emoticonEnable` should be checked at first in the `if` condition since regex match has cost
2. `node.value` was already replaced at L40 so the condition `RE_EMOJI.test(node.value)` always does not match. Instead, checking if actually something was replaced or not by comparing a string before/after replace.

Reply:
Good idea. I've changed it as requested.

Reply:
Oh, there is a test issue but I don't get it. Why is it failing? Your code proposal shouldn't change anything.
I'll have a closer look at it later today.